packages/react-reconciler/src/ReactFiberTreeReflection.js

```diff
@@ -342,20 +342,45 @@ export function isFiberSuspenseAndTimedOut(fiber: Fiber): boolean {
   );
 }
 
-// This is only safe to call in the commit phase when the return tree is consistent.
-// It should not be used anywhere else. See PR #18609 for details.
-export function isFiberInsideHiddenOrRemovedTree(fiber: Fiber): boolean {
-  let node = fiber;
-  let lastChild = null;
+function doesFiberContain(parentFiber: Fiber, childFiber: Fiber): boolean {
+  let node = childFiber;
+  const parentFiberAlternate = parentFiber.alternate;
   while (node !== null) {
-    if (
-      node.effectTag & Deletion ||
-      (isFiberSuspenseAndTimedOut(node) && node.child === lastChild)
-    ) {
+    if (node === parentFiber || node === parentFiberAlternate) {
       return true;
     }
-    lastChild = node;
     node = node.return;
   }
   return false;
 }
+
+function isFiberInHiddenTreeThatsContainsTargetFiber(
```

Comment:
Maybe `isFiberTimedOutSuspenseThatContainsTargetFiber`?

I think I just understood what this is checking. This isn't meant to find whether a Fiber is "somewhere in a timed out tree", this is meant to check whether the Fiber is literally the `<Suspense>` node itself that has timed out.

Reply:
Yes.